src/test/infra/lttng/test_records_provider_lttng.py

```diff
@@ -378,6 +485,215 @@ def test_get_publisher_handles(self, mocker):
+                COLUMN_NAME.SOURCE_TIMESTAMP: 1,
+            }),
+            RecordCppImpl({
+                COLUMN_NAME.RMW_TAKE_TIMESTAMP: 6,
+                COLUMN_NAME.SOURCE_TIMESTAMP: 7,
+            }),
+        ]
+        take_records = RecordsCppImpl(
+            take_records_data,
+            [
+                ColumnValue(COLUMN_NAME.RMW_TAKE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.SOURCE_TIMESTAMP),
+            ]
+        )
+
+        mocker.patch.object(provider_mock, 'subscription_take_records', return_value=take_records)
+
+        records_data: list[RecordInterface]
+        pub_records_data = [
+            RecordCppImpl({
+                COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: 2,
+                COLUMN_NAME.RCL_PUBLISH_TIMESTAMP: 3,
+                COLUMN_NAME.DDS_WRITE_TIMESTAMP: 4,
+                COLUMN_NAME.MESSAGE_TIMESTAMP: 5,
+                COLUMN_NAME.SOURCE_TIMESTAMP: 6,
+            }),
+            RecordCppImpl({
+                COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: 8,
+                COLUMN_NAME.RCL_PUBLISH_TIMESTAMP: 9,
+                COLUMN_NAME.DDS_WRITE_TIMESTAMP: 10,
+                COLUMN_NAME.MESSAGE_TIMESTAMP: 11,
+                COLUMN_NAME.SOURCE_TIMESTAMP: 12,
+            }),
+        ]
+
+        pub_records = RecordsCppImpl(
+            pub_records_data,
+            [
+                ColumnValue(COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.RCL_PUBLISH_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.DDS_WRITE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.MESSAGE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.SOURCE_TIMESTAMP),
+            ]
+        )
+        pub_records.rename_columns({
+            COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: (
+                f'/{COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP}'
+                )
+            })
+
+        mocker.patch.object(provider_mock, 'publish_records', return_value=pub_records)
+        mocker.patch.object(node_path_mock, 'publish_topic_name', '')
+
+        node_records = NodeRecordsUseLatestMessage(provider_mock, node_path_mock)
+        records = node_records.to_records()
+
+        expect_data = [
+            RecordCppImpl({
+                COLUMN_NAME.SOURCE_TIMESTAMP: 1,
+                COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: 2,
+            }),
+            RecordCppImpl({
+                COLUMN_NAME.SOURCE_TIMESTAMP: 7,
+                COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: 8,
+            }),
+        ]
+        expect_records = RecordsCppImpl(
+            expect_data,
+            [
+                ColumnValue(COLUMN_NAME.SOURCE_TIMESTAMP),
+                ColumnValue(COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP),
+            ]
+        )
+        expect_records.rename_columns({
+            COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP: (
+                f'/{COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP}'
+            )
+        })
+        records.equals(expect_records)
```

Comment:
Please add `assert`.
```suggestion
        assert records.equals(expect_records)
```